src/api/instance/instance.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+import {type CredrankInput} from "../credrank";
+import {CredGraph} from "../../core/credrank/credGraph";
+import {Ledger} from "../../core/ledger/ledger";
+import {type DependenciesConfig} from "../dependenciesConfig";
+
+/**
+This interface represents a SourceCred Instance, which is simply a read/write
+interface for configurations and outputs of the main SourceCred API.
+ */
+export interface Instance {
```

Comment:
At a glance, it’s not clear to me why this API has the shape that it
does. It’s not very orthogonal: the CredRank input is read only, the
ledger and dependencies config are write-only, and the cred graph is
read-write. As a reader, I’m left wondering why.

Diving in a bit deeper, we can see that a ledger and a dependencies
config are part of the CredRank input, so that explains why they don’t
need dedicated read methods, but the factoring still feels lopsided. And
the cred graph I/O methods are actually a bit different than the rest,
since CredRank never needs to read the cred graph; presumably this is
intended for downstream consumers, but this isn’t clear.

What do you think about an interface that more closely mirrors the
structure of CredRank, and clarifies how `readCredGraph` is different
from the rest? Something like:

```javascript
interface Instance {
  /** Reads inputs required to run CredRank. */
  readCredrankInput(): Promise<CredrankInput>;
  /** Writes output after running CredRank. */
  writeCredrankOutput(o: CredrankOutput): Promise<void>;

  /** Reads a cred graph generated by a previous run of CredRank. */
  readCredGraph(): Promise<CredGraph>;
}
```

Reply:
The reason it's lopsided for now is that I only added the minimum needed for credrank, but plan to expand it for other commands/usage. I'm down to combine the writes like you suggest right now.